pshtt/pshtt.py

```diff
@@ -654,88 +685,51 @@ def https_check(endpoint):
-        utils.debug("{}: {}".format(endpoint.url, err))
-        return
-
-    # Debugging
-    # for msg in cert_response:
-    #     print(msg)
-
     # Default endpoint assessments to False until proven True.
     endpoint.https_expired_cert = False
     endpoint.https_self_signed_cert = False
     endpoint.https_bad_chain = False
     endpoint.https_bad_hostname = False
 
-    # STORE will be either "Mozilla" or "Custom"
-    # depending on what the user chose.
-
-    # A certificate can have multiple issues.
-    for msg in cert_response:
-
-        # Check for missing SAN.
-        if (
-            (("DNS Subject Alternative Names") in msg) and
-            (("[]") in msg)
-        ):
-            endpoint.https_bad_hostname = True
-
-        # Check for certificate expiration.
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("certificate has expired") in msg)
-        ):
-            endpoint.https_expired_cert = True
+    cert_chain = cert_plugin_result.certificate_deployments[0].received_certificate_chain
+    leaf_cert = cert_chain[0]
+
+    # Check for leaf certificate expiration/self-signature.
+    if leaf_cert.not_valid_after < datetime.datetime.now():
+        endpoint.https_expired_cert = True
+
+    # Check to see if the cert is self-signed
+    if leaf_cert.issuer is leaf_cert.subject:
+        endpoint.https_self_signed_cert = True
+
+    # Check certificate chain
+    # NOTE: If this is the only flag that's set, it's probably
+    # an incomplete chain
+    # If this isn't the only flag that is set, it might be
+    # because there is another error. More debugging would
+    # need to be done at this point, but not through sslyze
+    # because sslyze doesn't have enough granularity
+    for cert in cert_chain[1:]:
+        # Check for certificate expiration
+        if cert.not_valid_after < datetime.datetime.now():
+            endpoint.https_bad_chain = True
 
         # Check to see if the cert is self-signed
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("self signed certificate") in msg)
-        ):
-            endpoint.https_self_signed_cert = True
-
-        # Check to see if there is a bad chain
-
-        # NOTE: If this is the only flag that's set, it's probably
-        # an incomplete chain
-        # If this isnt the only flag that is set, it's might be
-        # because there is another error. More debugging would
-        # need to be done at this point, but not through sslyze
-        # because sslyze doesn't have enough granularity
-
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (
-                (("unable to get local issuer certificate") in msg) or
-                (("self signed certificate") in msg)
-            )
-        ):
+        if cert.issuer is (cert.subject or None):
```

Comment:
Same comment as above about usage of `is`. I'm not a big fan of a `cert.issuer == None` possibility, so it could be refactored.

```suggestion
        if cert.issuer == (cert.subject or None):
```